src/167570.c

```diff
@@ -115,9 +118,226 @@ void filemenu_yesno_init(MenuPanel* tab) {
     tab->initialized = TRUE;
 }
 
-static const f32 padding = 0.0f;
+void filemenu_yesno_handle_input(MenuPanel* menu) {
+    s32 oldSelected = menu->selected;
 
-INCLUDE_ASM(s32, "167570", filemenu_yesno_handle_input);
+    if (filemenu_heldButtons & BUTTON_STICK_UP) {
+        menu->row--;
+        if (menu->row < 0) {
+            menu->row = 0;
+        }
+    }
+
+    if (filemenu_heldButtons & BUTTON_STICK_DOWN) {
+        menu->row++;
+        if (menu->row >= menu->numRows) {
+            menu->row = menu->numRows - 1;
+        }
+    }
+
+    menu->selected = MENU_PANEL_SELECTED_GRID_DATA(menu);
+
+    if (oldSelected != menu->selected) {
+        sfx_play_sound(SOUND_MENU_CHANGE_SELECTION);
+    }
+
+    if ((filemenu_pressedButtons & BUTTON_START) && menu->page == 4) {
+        filemenu_set_selected(menu, 0, 0);
+        filemenu_pressedButtons = BUTTON_A;
+    }
+
+    if (filemenu_pressedButtons & BUTTON_A) {
+        u32 slot;
+        u32 slot2;
+        u32 slot3;
+        u32 slot4;
+        s32 i;
+
+        sfx_play_sound(SOUND_MENU_NEXT);
+
+        switch (menu->selected) {
+            case 0:
+                switch (menu->page) {
+                    case 0:
+                        filemenu_8024C098 = 2;
+                        filemenu_menus[filemenu_8024C098]->page = 0;
+                        gWindows[47].width = 182;
+                        gWindows[47].height = 25;
+                        gWindows[47].pos.x = (gWindows[47].parent != -1) ?
+                                             (gWindows[gWindows[47].parent].width / 2) - 91 :
+                                             69;
+                        gWindows[47].pos.y = ((gWindows[47].parent != -1) ?
+                                             (gWindows[gWindows[47].parent].height / 2) : 120) - gWindows[47].height / 2;
```

Comment:
These ternaries are hard to read imo, I'd write them as something like
```suggestion
                        gWindows[47].pos.x = gWindows[47].parent != -1
                            ? gWindows[gWindows[47].parent].width / 2) - 91
                            : 69;
                        gWindows[47].pos.y = (
                            gWindows[47].parent != -1
                                ? (gWindows[gWindows[47].parent].height / 2)
                                : 120
                        ) - gWindows[47].height / 2;
```